README.md

```diff
@@ -21,49 +73,409 @@ that's been made default from that version forward.
+setup with webpacker).
+
+The installer will add TypeScript and Angular core libraries using yarn plus
+any changes to the configuration files. An example component written in
+TypeScript will also be added to your project in `app/javascript` so that
+you can experiment with Angular right away.
+
+
+### Vue
+
+To use Webpacker with [Vue](https://vuejs.org/), create a
+new Rails app using new `--webpack=vue` option:
+
+```bash
+# Rails 5.1+
+./bin/rails new myapp --webpack=vue
+```
+(or run `./bin/rails webpacker:install:vue` on a Rails app already setup with webpacker).
+
+The installer will add Vue and required libraries using yarn plus
+any changes to the configuration files. An example component will
+also be added to your project in `app/javascript` so that you can
+experiment Vue right away.
+
+
+### Elm
+
+To use Webpacker with [Elm](http://elm-lang.org), create a
+new app using new `--webpack=elm` option:
+
+```
+./bin/rails new myapp --webpack=elm
+```
+
+(or run `./bin/rails webpacker:install:elm` on a Rails app already setup with webpacker).
+
+The Elm library and core packages will be added via Yarn and Elm itself.
+An example `Main.elm` app will also be added to your project in `app/javascript`
+so that you can experiment with Elm right away.
+
 
 ## Binstubs
 
 Webpacker ships with two binstubs: `./bin/webpack` and `./bin/webpack-dev-server`.
-They're thin wrappers around the standard webpack.js executable, just to ensure that the right configuration file is loaded depending on your environment.
+They're thin wrappers around the standard `webpack.js` and `webpack-dev-server.js`
+executable to ensure that the right configuration file is loaded
+depending on your environment.
+
+
+### Webpack dev server
+
+In development, you'll need to run `./bin/webpack-dev-server` in a separate terminal
+from `./bin/rails server` to have your `app/javascript/packs/*.js` files compiled
+as you make changes.
+
+`./bin/webpack-dev-server` launches the [Webpack Dev Server](https://webpack.js.org/configuration/dev-server/), which serves your pack files
+on `http://localhost:8080/` by default and supports live code reloading in development environment. You will need to install additional plugins for Webpack if you want features like [Hot Module Replacement](https://webpack.js.org/guides/hmr-react/)
+
+If you'd rather not have to run the two processes separately by hand, you can use [Foreman](https://ddollar.github.io/foreman):
+
+```bash
+gem install foreman
+```
+
+```yml
+# Procfile
+web: bundle exec rails s
+webpacker: ./bin/webpack-dev-server
+```
+
+```bash
+foreman start
+```
+
+You can also pass additional cli options supported by [webpack-dev-server](https://webpack.js.org/configuration/dev-server/). Please note that inline
+options will always take precedence over the ones
+already set in the configuration file.
+
+```bash
+./bin/webpack-dev-server --host 0.0.0.0 --inline true --hot false
+```
+
+
+### Webpack watcher
```

Comment:
Doesn't feel right to call this "Webpack watcher". I'd just document the existence of `bin/webpack` and that you can pass options like `--watch`.